lib/ansiblelint/__init__.py

```diff
@@ -50,53 +50,76 @@ def matchlines(self, file, text):
         # arrays are 0-based, line numbers are 1-based
         # so use prev_line_no as the counter
         for (prev_line_no, line) in enumerate(text.split("\n")):
+            rule_id_list = ansiblelint.utils.get_rule_skips_from_line(line)
+            if self.id in rule_id_list:
+                continue
+
             result = self.match(file, line)
-            if result:
-                message = None
-                if isinstance(result, str):
-                    message = result
-                matches.append(Match(prev_line_no+1, line,
-                               file['path'], self, message))
+            if not result:
+                continue
+            message = None
+            if isinstance(result, str):
+                message = result
+            matches.append(Match(prev_line_no+1, line,
+                           file['path'], self, message))
         return matches
 
     def matchtasks(self, file, text):
```

Comment:
Not strictly related to the current change, but `file` should be `file_` or something that doesn't shadow builtins.

Reply:
Opened #469 to address this